internal/processors/cloudeventconvert/attesation_msg.go

```diff
@@ -80,11 +80,11 @@ func (c *cloudeventProcessor) verifySignature(event *cloudevent.CloudEvent[json.
 	}
 
 	signature := common.FromHex(sig)
-	msgHash := crypto.Keccak256Hash(event.Data)
+	msgHashWithPrfx := crypto.Keccak256Hash([]byte(fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(event.Data), event.Data)))
```

Comment:
I'm gonna put in another plug for [`TextHash`](https://pkg.go.dev/github.com/ethereum/go-ethereum/accounts#TextHash) here. It's the same code, but you get to offload the responsibility.